### PR TITLE
Create skeleton for package-json-lint project

### LINKED ISSUE #538: Create skeleton for package-json-lint package
# Story

## Description

As a developer of terra, I want to have a skeleton package package-json-lint that will be used to flag problems that are present in a package.json

## Motivation

So that I can start developing that linter for teams

## Definition of Done

We'll know we're done when:

* The package is created is merged to main and the overall build for terra-toolkit succeeds
* The package is not configured to release

### PR DESCRIPTION
**Pull request by ryanthemanuel**
### Summary

This is just the initial commit of the package-json-lint project. It is set up not to be released and will be built out as a part of #540 

Fixes #538

### CODE REVIEW

**Review by StephenEsser — APPROVED** [review]

File: packages/package-json-lint/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+1. MAJOR versions represent breaking changes
+2. MINOR versions represent added functionality in a backwards-compatible manner
+3. PATCH versions represent backwards-compatible bug fixes
+
+Consult the component CHANGELOGs, related issues, and PRs for more information.
+
+## LICENSE
+
+Copyright 2020 Cerner Innovation, Inc.
```
Comment (ryanthemanuel) [inline packages/package-json-lint/README.md]:
```suggestion
Copyright 2021-present Cerner Innovation, Inc.
```

File: packages/package-json-lint/LICENSE [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2020 Cerner Innovation, Inc.
```
Comment (ryanthemanuel) [inline packages/package-json-lint/LICENSE]:
```suggestion
   Copyright 2021-present Cerner Innovation, Inc.
```

File: packages/package-json-lint/NOTICE [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Copyright 2020 Cerner Innovation, Inc.
```
Comment (ryanthemanuel) [inline packages/package-json-lint/NOTICE]:
```suggestion
Copyright 2021-present Cerner Innovation, Inc.
```

File: packages/package-json-lint/.npmignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
+*.log
+target
+reports
+tests/**/__snapshots__/
+aggregated-translations
+dev-site-config
```
Comment (mjhenkes) [inline packages/package-json-lint/.npmignore]: i assume you don't need these

File: packages/package-json-lint/package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    "Cerner"
+  ],
+  "author": "Cerner Corporation",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/cerner/terra-toolkit/issues"
+  },
+  "engines": {
+    "node": "^10.13.0"
+  },
```
Comment (mjhenkes) [inline packages/package-json-lint/package.json]: do we need this?

File: packages/package-json-lint/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+## Versioning
+
+package-json-lint is considered to be stable and will follow [SemVer](http://semver.org/) for versioning.
+
+1. MAJOR versions represent breaking changes
+2. MINOR versions represent added functionality in a backwards-compatible manner
+3. PATCH versions represent backwards-compatible bug fixes
+
+Consult the component CHANGELOGs, related issues, and PRs for more information.
```
Comment (emilyrohrbough) [inline packages/package-json-lint/README.md:16]: Can this be removed since it's describe at the root level?

File: lerna.json [hunk 7]
```diff
@@ -3,6 +3,7 @@
   "command": {
     "publish": {
       "ignoreChanges": [
+        "packages/package-json-lint/**",
```
Comment (emilyrohrbough) [inline lerna.json:6]: Should this be package-json-linter? instead of just lint?

File: packages/package-json-lint/README.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# package.json Linter
+
+[![NPM version](https://badgen.net/npm/v/@cerner/package-json-lint)](https://www.npmjs.org/package/@cerner/package-json-lint)
+[![Build Status](https://badgen.net/travis/cerner/terra-toolkit)](https://travis-ci.com/cerner/terra-toolkit)
+
+Linter for package.json files
```
Comment (emilyrohrbough) [inline packages/package-json-lint/README.md]:
```suggestion
Linter for package.json files.
```

File: packages/package-json-lint/package.json [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    "access": "public"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cerner/terra-toolkit.git",
+    "directory": "packages/package-json-lint"
+  },
+  "keywords": [
+    "lint",
+    "package.json",
```
Comment (emilyrohrbough) [inline packages/package-json-lint/package.json]: I'm not sure package.json is needed s a keyword

File: lerna.json (hunk 7 above)
Comment (ryanthemanuel) [reply lerna.json:6]: It's eslint and stylelint not eslinter or stylelinter.  I was trying to follow that pattern.

File: packages/package-json-lint/README.md (hunk 6 above)
Comment (ryanthemanuel) [reply packages/package-json-lint/README.md:16]: I was mimicking existing packages. As I think about it, it wouldn't show up in the npm page if we didn't include that (if that matters): e.g. https://www.npmjs.com/package/@cerner/terra-cli

File: packages/package-json-lint/package.json [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    "package-json-lint",
+    "Cerner"
+  ],
+  "author": "Cerner Corporation",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/cerner/terra-toolkit/issues"
+  },
+  "engines": {
+    "node": "^10.13.0"
```
Comment (ryanthemanuel) [inline packages/package-json-lint/package.json]: Spoke with @mjhenkes and we landed on:

```suggestion
    "node": ">=10"
```

File: packages/package-json-lint/.npmignore (hunk 4 above)
Comment (ryanthemanuel) [inline packages/package-json-lint/.npmignore]:
```suggestion
```

Comment (ryanthemanuel) [reply packages/package-json-lint/.npmignore]: Probably right.

File: packages/package-json-lint/package.json (hunk 9 above)
Comment (ryanthemanuel) [inline packages/package-json-lint/package.json]:
```suggestion
```

Comment (ryanthemanuel) [reply packages/package-json-lint/package.json]: I'm fine removing this.

**Review by mjhenkes — APPROVED** [review]

**Review by benbcai — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**